### DISCUSSION TITLE
[Itau] Cadastrar webhook retornando erro 500

### DISCUSSION THREAD

**Discussion by mattshuegel** [bacen/pix-api#411] [General]
Pessoal, boa noite.

Estou enfrentando um problema ao fazer o cadastramento do webhook junto ao Itaú, ao submeter minha URL para eles, estão retornando como: 
```
{
    "codigo": "500",
    "mensagem": "Erro interno da API. Favor contatar equipe técnica Itaú Unibanco."
}
```

Minha implementação está sendo feita com dotnet core. A mesma está sendo encapsulada em um container docker que por sua vez, é acessado através do nginx.

Já fizemos a instalação do cadeia de certificados do Itaú no servidor, porém permanece o erro.

Alguém já passou por isso? Ou alguém sabe como resolver?

**Comment by clevison** [comment]
Tive o mesmo problema, A api do Webhook tem algumas limitações, a primeira é que você não pode cadastrar mais de uma URL da sua aplicação e a segunda é que as chaves de cadastro do webhook só estão funcionando via CNPJ;

Para cadastrar envie um PUT para a URL:
`URL_API_ITAU_PROD/pix_recebimentos/v2/webhook/SEU_CNPJ
`Com o Body
```
{
    "webhookUrl":"sua_url"
}
```

**Reply by rubenskuhl** [reply]
> Tive o mesmo problema, A api do Webhook tem algumas limitações, a primeira é que você não pode cadastrar mais de uma URL da sua aplicação e a segunda é que as chaves de cadastro do webhook só estão funcionando via CNPJ;
> 
> Para cadastrar envie um PUT para a URL: `URL_API_ITAU_PROD/pix_recebimentos/v2/webhook/SEU_CNPJ `Com o Body
> 
> ```
> {
>     "webhookUrl":"sua_url"
> }
> ```

Isso é totalmente fora do padrão da API. Já atualizei o #76 com essa triste informação.

**Reply by clevison** [reply]
tecnicamente na api é possível é possível cadastrar vários webhooks com a chave que você quiser, porém tem uma nota na doc informando que a chave precisa ser o CNPJ e entramos em contato com a equipe deles e fomos informados que ao cadastrar mais de uma URL no Webhook só é considerada a última que foi cadastrada, isso é um bug e estão tentando resolver.
Concordo que levando em conta o que temos de documentação do Webhook, isso foge bastante da documentação.

**Reply by vitorvmax** [reply]
> Para cadastrar envie um PUT para a URL:
> `URL_API_ITAU_PROD/pix_recebimentos/v2/webhook/SEU_CNPJ `Com o Body

Olá meu amigo, sabe me dizer se até hoje isso acontece (apenas chave CNPJ)? Grato!
Eu consegui cadastrar uma chave PIX aleatória, mas só funcionou uma vez (retornou status code 201). Agora eu não consigo listar os Webhooks, deletar ou cadastrar outro webhookUrl.
